tests/end_to_end/features/autovariants.feature

```diff
@@ -0,0 +1,47 @@
+Feature: AutoVariants
+    Scenario: Same auto variant for same transformation 
+        Given Featureform is installed
+        When I turn on autovariants
+        And I create a "hosted" "insecure" client for "localhost:7878"
+        And I register "s3" filestore with bucket "featureform-spark-testing" and root path "data"
+        And I register databricks
+        And I register transactions_short.csv
+        And I register a transformation with auto variant
+        Then I should be able to reuse the same variant for the same transformation
+        And I can get the transformation as df
+        And I turn off autovariants
+
+    Scenario: User-defined variant and auto variant for same transformation
+        Given Featureform is installed
+        When I turn on autovariants
+        And I create a "hosted" "insecure" client for "localhost:7878"
+        And I register "s3" filestore with bucket "featureform-spark-testing" and root path "data"
+        And I register databricks
+        And I register transactions_short.csv
+        And I register a transformation with user-provided variant
+        Then I should be able to register a new auto variant transformation
+        And I can get the transformation as df
+        And I turn off autovariants
+    
+    Scenario: Different auto variant for different transformation
+        Given Featureform is installed
+        When I turn on autovariants
+        And I create a "hosted" "insecure" client for "localhost:7878"
+        And I register "s3" filestore with bucket "featureform-spark-testing" and root path "data"
+        And I register databricks
+        When I register transactions_short.csv
+        And I register a transformation with auto variant
+        Then I should be able to register a modified transformation with new auto variant
```

Comment:
a bit of a vent but maybe you share the same opinion: I don't like that behave tests are just becoming glorified comments for code. I think having a definitions file with asserts might be better. I've been finding it even slightly difficult to navigate to see what the previous step is doing, often times I'd rather just look at code.

Are we going to register the same databricks for each one, the health check has to run and it's going to take awhile? should we use the same one?

Reply:
I'm not advocating you completely change how things are done here but just saying with the path a lot of these behave tests are going it might be better to do what sterling talked about and shove definitions file into the behave framework